src/firewall/core/fw_zone.py

```diff
@@ -200,7 +200,7 @@ def zone_from_chain(self, chain):
                 _chain = x
         if _chain is not None:
             # next part needs to be zone name
-            if splits[1] not in self.get_zones():
+            if self.get_zone(splits[1], required=False) is None:
```

Comment:
I still don't like this `required` keyword argument. I'd rather see a different function, e.g. `zone_exists()`.

Reply:
we have `check_zone()`, `get_zone()`, `get_zones()`, that work together.

Yes, there could be also a `has_zone()` (`zone_exists()`)...

Then
- `has_zone(zone_name)` is like `bool(get_zone(zone_name, required=False))`
- or
```python
try:
    self.check_zone(zone_name)
    has = True
except FirewallException:
    has = False
```

I don't mind. I am just not sure that having another variant on this.

Reply:
there is now `has_{zone,policy,ipset}()`, which follows the naming patterns with `{get,query,check,has}_OBJ()`